### PR TITLE
fix: href-clickable

### PR DESCRIPTION
**Pull request by MarcoKorinth**
# Changes
HTML rendering was adjusted so that `<a>` tags are only clickable if one of the following conditions apply:
- The url starts with `#`, `mailto`, `tel`, `http` or `ftp`
- The url starts with `data` and has mimetype for `png` or `jpeg`

### CODE REVIEW

File: app/cypress/e2e/SecvisogramPage/SecureHref.cy.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import HTMLTemplate from '../../../lib/app/SecvisogramPage/View/shared/HTMLTemplate.js'
+import DocumentEntity from '../../../lib/app/shared/Core/entities/DocumentEntity.js'
+import secureHrefTests from '../../fixtures/secureHrefTests.js'
+
+describe('secureHrefTests', () => {
+  secureHrefTests.forEach((test, i) => {
+    it(test.title ?? `Test #${i + 1}`, () => {
+      const preview = new DocumentEntity().preview({ document: test.content })
+      const html = HTMLTemplate(preview)
+      const hasHref = html.match(/<a href=".*test\.org.*">.*<\/a>/)
```
Comment (tschmidtb51) [inline app/cypress/e2e/SecvisogramPage/SecureHref.cy.js:11]: Please use example.com instead - test.org is IMHO not reserved for test purposes...
```suggestion
      const hasHref = html.match(/<a href=".*example\.com.*">.*<\/a>/)

File: app/cypress/fixtures/secureHrefTests.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      },
+    },
+  },
+  {
+    title: 'data href (MIME-type png) (valid)',
+    valid: true,
+    content: {
+      document: {
+        ...minimalDoc.document,
+        acknowledgments: [{ urls: ['data:image/png;base64,iVBORwtest.org0KGgoAAAANSUhEUgAAA'] }],
```
Comment (tschmidtb51) [inline app/cypress/fixtures/secureHrefTests.js:71]: Hmm - that doesn't look like a valid base64 to me. But the idea is good  - we should cover both cases (invalid and valid base64).

File: app/cypress/fixtures/secureHrefTests.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    valid: false,
+    content: {
+      document: {
+        ...minimalDoc.document,
+        acknowledgments: [{ urls: ['wss://test.org'] }],
+      },
+    },
+  },
+  {
+    title: 'data href (MIME-type gif) (invalid)',
```
Comment (tschmidtb51) [inline app/cypress/fixtures/secureHrefTests.js:118]: I like the idea. We should also put that on the "allowed" list. But SVG should be on the block list.

File: app/lib/app/SecvisogramPage/View/shared/HTMLTemplate/Template.html [hunk 4]
```diff
@@ -331,7 +331,7 @@ <h2>Sharing rules</h2>
     <p>
       {{#tlp}}
         <b>{{#label}}TLP:{{.}}{{/label}}</b><br>
-        For the TLP version see: <a href={{#url}}{{.}}{{/url}}{{^url}}https://www.first.org/tlp/{{/url}}>{{#url}}{{.}}{{/url}}{{^url}}https://www.first.org/tlp/{{/url}}</a>
+        For the TLP version see: <a {{#secureHref}}{{#url}}{{.}}{{/url}}{{/secureHref}}{{^url}}https://www.first.org/tlp/{{/url}}>{{#url}}{{.}}{{/url}}{{^url}}https://www.first.org/tlp/{{/url}}</a>
```
Comment (tschmidtb51) [inline app/lib/app/SecvisogramPage/View/shared/HTMLTemplate/Template.html:334]: Actually, the current implementation breaks the default value (`https://www.first.org/tlp/`) that is inserted if no `url` is given. I guess the following might fix it:
```suggestion
        For the TLP version see: <a {{#secureHref}}{{#url}}{{.}}{{/url}}{{/secureHref}}{{^url}}href=https://www.first.org/tlp/{{/url}}>{{#url}}{{.}}{{/url}}{{^url}}https://www.first.org/tlp/{{/url}}</a>

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
Please see my comments.

File: app/cypress/e2e/SecvisogramPage/SecureHref.cy.js [hunk 5]
```diff
@@ -8,8 +8,9 @@ describe('secureHrefTests', () => {
     it(test.title ?? `Test #${i + 1}`, () => {
       const preview = new DocumentEntity().preview({ document: test.content })
       const html = HTMLTemplate(preview)
-      const hasHref = html.match(/<a href=".*test\.org.*">.*<\/a>/)
+      const url = test.url.replaceAll('/', '&#x2F;').replaceAll('=', '&#x3D;')
+      const hasHref = html.match(new RegExp(`<a href="${url}">.*</a>`))
```
Comment (tschmidtb51) [inline app/cypress/e2e/SecvisogramPage/SecureHref.cy.js:12]: Please add some comments here.

File: app/cypress/fixtures/secureHrefTests.js [hunk 6]
```diff
@@ -5,80 +5,107 @@ export default [
...
         ...minimalDoc.document,
-        acknowledgments: [{ urls: ['https://test.org'] }],
+        acknowledgments: [{ urls: ['https://example.com'] }],
       },
     },
   },
   {
     title: 'mailto href (valid)',
     valid: true,
+    url: 'mailto://user@example.com',
```
Comment (tschmidtb51) [inline app/cypress/fixtures/secureHrefTests.js:41]: `mailto:` usually does not have the `//`
But happy to test both...

File: app/cypress/fixtures/secureHrefTests.js [hunk 7]
```diff
@@ -5,80 +5,107 @@ export default [
...
         ...minimalDoc.document,
-        acknowledgments: [{ urls: ['mailto://user@test.org'] }],
+        acknowledgments: [{ urls: ['mailto://user@example.com'] }],
       },
     },
   },
   {
     title: 'tel href (valid)',
     valid: true,
+    url: 'tel://\\+4934587913248',
```
Comment (tschmidtb51) [inline app/cypress/fixtures/secureHrefTests.js:52]: `tel:` usually does not have the `//`
But happy to test both...

File: app/cypress/fixtures/secureHrefTests.js [hunk 8]
```diff
@@ -5,80 +5,107 @@ export default [
...
     content: {
       document: {
         ...minimalDoc.document,
-        acknowledgments: [{ urls: ['tel://+4934587913248test.org'] }],
+        acknowledgments: [{ urls: ['tel://+4934587913248'] }],
```
Comment (tschmidtb51) [inline app/cypress/fixtures/secureHrefTests.js:56]: We can't use a real number here. I checked with the authority, we can use any of those described in https://www.bundesnetzagentur.de/SharedDocs/Downloads/DE/Sachgebiete/Telekommunikation/Unternehmen_Institutionen/Nummerierung/Rufnummern/mittlg148_2021.pdf?__blob=publicationFile&v=1

=> +493023125000

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
Some more comments

**Review by tschmidtb51 — APPROVED** [review]
LGTM

**Review by StefanNemeth — APPROVED** [review]

**Pull request merged**